pydra/engine/task.py

```diff
@@ -385,7 +400,13 @@ def __init__(
     @property
     def cmdline(self):
         """Get the actual command line that will be submitted."""
-        return " ".join(self.container_args + self.command_args)
+        orig_inputs = attr.asdict(self.inputs)
+        modified_inputs = template_update(self.inputs)
+        if modified_inputs is not None:
+            self.inputs = attr.evolve(self.inputs, **modified_inputs)
+        cmdline = " ".join(self.container_args + self.command_args)
+        self.inputs = attr.evolve(self.inputs, **orig_inputs)
```

Comment:
was this needed to pass the existing tests?

Reply:
this was updated to reflect the fact that one can call `cmdline`  before calling run. and this property is not used for running.

but to answer your original question, i don't think there was a test that tested this.

Reply:
I'm trying to find situation when your modification changes anything. If you think about a single task and running `cmdline` before `run`, this worked without your changes. If you think about workflows, this doesn't work, there are still `LF`.  Could you give me some examples or revert the changes for now.

Reply:
if you prefer to keep it, I'll just make a note to think about it later, but couldn't find a good example

Reply:
you will also have to take the changes away from `run`. to test the issue, take the current master branch, print `cmdline` from a container task, run it, and then print `cmdline` again.

Reply:
in master this modified test will fail. (i'll push it to this branch)

```python
def test_docker_inputspec_3(plugin, tmpdir):
    """ input file is in the container, so metadata["container_path"]: True,
        the input will be treated as a str """
    filename = "/proc/1/cgroup"

    cmd = "cat"

    my_input_spec = SpecInfo(
        name="Input",
        fields=[
            (
                "file",
                File,
                dc.field(
                    metadata={
                        "mandatory": True,
                        "position": 1,
                        "help_string": "input file",
                        "container_path": True,
                    }
                ),
            )
        ],
        bases=(DockerSpec,),
    )

    docky = DockerTask(
        name="docky",
        image="busybox",
        executable=cmd,
        file=filename,
        input_spec=my_input_spec,
        strip=True,
    )

    cmdline = docky.cmdline # in master this will fail
    res = docky()
    assert "docker" in res.output.stdout
    assert cmdline == docky.cmdline
```

Reply:
I really don't see a difference between the old `cmdline` and the new one: you can see: https://github.com/satra/pydra/pull/12